src/clusterfuzz/_internal/bot/tasks/cron/batch_fuzzer_jobs.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A cron task that batches FuzzerJobs."""
+
+from google.cloud import ndb
+
+from clusterfuzz._internal.datastore import data_types
+
+FUZZER_JOB_BATCH_SIZE = 4000
+
+
+def batch_fuzzer_jobs():
+  """Batches FuzzerJobs for reduced Datastore read ops by bots."""
+  platforms = [
+      item.platform for item in data_types.FuzzerJob.query(
+          projection=[data_types.FuzzerJob.platform], distinct=True)
+  ]
+
+  for platform in platforms:
+    fuzzer_jobs = list(
+        data_types.FuzzerJob.query(data_types.FuzzerJob.platform == platform))
+    fuzzer_jobs.sort(key=lambda item: item.job)
+
+    batches_to_remove = set(
+        b.key for b in data_types.FuzzerJobs.query(
+            data_types.FuzzerJobs.platform == platform))
+
+    batch_count = 0
+    for i in range(0, len(fuzzer_jobs), FUZZER_JOB_BATCH_SIZE):
+      key_id = platform + '-' + str(batch_count)
```

Comment:
`f-string`.

Reply:
I copied this from the original code and will leave this unchanged first to validate migration.